### PR TITLE
Add step to clean up Filestore instances created by e2e jobs

### PR DESCRIPTION
**Pull request by bertinatto**
Also, remove periodic jobs until the job becomes stable (not all tests are currently passing).

CC @openshift/storage

### CODE REVIEW

File: ci-operator/config/openshift/gcp-filestore-csi-driver-operator/openshift-gcp-filestore-csi-driver-operator-main.yaml [hunk 1]
```diff
@@ -102,6 +102,10 @@ resources:
       memory: 200Mi
 test_binary_build_commands: make test
 tests:
+- as: unit
+  commands: make test-unit
+  container:
+    from: src
```
Comment (dobsonj) [inline ci-operator/config/openshift/gcp-filestore-csi-driver-operator/openshift-gcp-filestore-csi-driver-operator-main.yaml:108]: Do you want this in the other branches too, or just `main` for now?

Comment (bertinatto) [reply ci-operator/config/openshift/gcp-filestore-csi-driver-operator/openshift-gcp-filestore-csi-driver-operator-main.yaml:108]: Fixed!

**Pull request merged**